### DISCUSSION TITLE
Unable to display rule HTML data (field.rulesHtmlData)

### DISCUSSION THREAD

**Discussion by franck-thorck** [solspace/craft-freeform#1042] [Questions & Answers] [answered]
**Describe the bug or issue you're experiencing**

I have update from FreeForm version 3 to 4, and from 4 to 5, everything works correctly. But I setuped conditions/rules for some fields and try to use `rulesHtmlData` property, and it dumps empty value for hidden field.

**Steps to reproduce**

1. Set conditional rule for a field
2. Use field.rulesHtmlData in templating

**Expected behavior**

Should parses the rule for html rendering as the doc explains it: [https://docs.solspace.com/craft/freeform/v5/templates/objects/field/#custom-attributes-customattributes](https://docs.solspace.com/craft/freeform/v5/templates/objects/field/#custom-attributes-customattributes)

**Craft & Plugin Info (please complete the following information):**

- Craft Version: 4.7.1
- Freeform Version: 5.0.8
- Freeform Edition: Pro
- Fresh Install or Upgrade: Upgrade

**Additional context**

<img width="507" alt="Capture d’écran, le 2024-02-07 à 17 31 19" src="https://github.com/solspace/craft-freeform/assets/117709151/df1a9afa-8e2b-4492-974c-794a4d19ad2f">
<img width="867" alt="Capture d’écran, le 2024-02-07 à 17 37 55" src="https://github.com/solspace/craft-freeform/assets/117709151/f71a1d8d-6f73-42b0-a4aa-dbbf89640ee6">

**Comment by kjmartens** [comment, marked as answer]
Hi @franck-thorck,

Sorry about the trouble here - we're still working through some remaining updates to the documentation. The process is different now for conditional rules. It now relies on an attribute that typically gets automatically inserted (`data-field-container="myFieldHandle"`).

If you need to do something more manual, we just set up a guide for it when it comes to manually templating forms:
https://docs.solspace.com/craft/freeform/v5/forms/conditional-rules/#templating

However, I believe in your case, you will need to check out the Template Overrides feature:
https://docs.solspace.com/craft/freeform/v5/templates/formatting/#template-overrides

And if you wish to have a custom formatting template, you can use something like the [Flexbox formatting template](https://docs.solspace.com/craft/freeform/v5/templates/formatting/flexbox/) (or any of the other ones) as a starting point. 🙂
